### DISCUSSION TITLE
no i'm actually scared not kidding

### DISCUSSION THREAD

**Discussion by IMGROOT2** [For-0/vocabustudy#183] [General]
![image](https://user-images.githubusercontent.com/116324098/197462327-1678d17c-4c9d-41c7-8c8f-e998fcfa5e80.png)

GitHub copilot just "guessed" that I helped design the apps and made the logo. WHAT ON EARTH HOW DID IT KNOW

**Comment by grimsteel** [comment]
Shut it down before it takes over Vocabustudy

**Reply by IMGROOT2** [reply]
LOL copilot is gonna replace programmers in like a decade

**Comment by grimsteel** [comment]
![image](https://user-images.githubusercontent.com/115331356/197634885-e585ab6a-3361-494c-95b9-bba4c17b4f94.png)
I'm not even going to ask

**Reply by IMGROOT2** [reply]
LMAOOOO I GOT ANAYA GUPTA

**Reply by grimsteel** [reply]
Do you know any of these people?

**Reply by grimsteel** [reply]
Are they even people?

**Reply by IMGROOT2** [reply]
They're probably robots ngl

**Reply by grimsteel** [reply]
True
I was trying to insert a tab character, so I accidentally accepted the suggestion...
I think that's what they wanted

**Reply by IMGROOT2** [reply]
We mustn't fall for their foolish tricks

**Reply by grimsteel** [reply]
Yes..
what I just navigated to the extensions tab in VS Code and Github Copilot was auto updating...
This has gone too far
![image](https://user-images.githubusercontent.com/115331356/197645225-f7d0784c-4338-4a63-bfe6-8dc391db7a3a.png)

**Reply by IMGROOT2** [reply]
Oh no it's too late, it's evolving

Also, is it possible to run the emulator directly from GitHub? I want to see the merged code from earlier

**Reply by grimsteel** [reply]
I could probably set up a "temporary preview url" action that would build the current code and generate a preview url for one hour. (That would use the production data, though...)
But apart from that, the only option is to use the preview urls in the PRs (those get updated for every commit)
https://vocabustudyonline--pr184-166-rearrange-elemen-nfq0hpjt.web.app

**Reply by grimsteel** [reply]
But then of course you could just `git fetch` and `git pull` while on main or the other branch (we can delete that branch now) then run the emulator.
Speaking of that, I forgot to run `git pull` myself... 😨 Luckily I haven't committed yet lol

**Reply by IMGROOT2** [reply]
> https://vocabustudyonline--pr184-166-rearrange-elemen-nfq0hpjt.web.app

This is perfect. So, do they automatically generate with pull requests? How do you get the links? I saw that you posted it in the UI Flow PR, but idk how you got it.

**Reply by grimsteel** [reply]
> I saw that you posted it in the UI Flow PR

I didn't - that was the bot that generates it

**Reply by IMGROOT2** [reply]
Ohhh okay so the bot automatically generates it when I make a PR. Got it ty

**Reply by grimsteel** [reply]
> So, do they automatically generate with pull requests

Yes

**Comment by grimsteel** [comment]
![image](https://user-images.githubusercontent.com/115331356/197653007-c307d697-09e2-4fe0-87c8-02d7c0510d59.png)
WHAT IT FIGURED OUT HOW TO USE MY OWN UTILITY FUNCTIONS
Also how did it know what I was going to do

**Reply by IMGROOT2** [reply]
LOL IT PREDICTED SO MUCH! is it actually what you were gonna do?

**Reply by grimsteel** [reply]
Yes...
We might have to change the credits to say that Github Copilot did most of the work

**Reply by IMGROOT2** [reply]
LMAOOO THAT WOULD BE... sad...

**Comment by grimsteel** [comment]
![image](https://user-images.githubusercontent.com/115331356/198142271-9bd8bb2f-0b92-4a17-a957-f05ef67d780c.png)
> Reload Required

That's what you think
